src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -336,6 +376,35 @@ public boolean shouldSkipMonitoredTripCheck() throws Exception {
                 if (!calculateNextItinerary()) return true;
             }
             LOG.info("Next itinerary happening on {}.", targetDate);
+
+            // update journey state with baseline departure and arrival times which are the last known departure/arrival
+            journeyState.baselineDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+            journeyState.baselineArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+
+            // update journey state with the original (scheduled departure and arrival times). Calculate these by
+            // finding the first/last transit legs and subtracting any delay.
+            journeyState.originalDepartureTimeEpochMillis = matchingItinerary.startTime.getTime();
+            for (Leg leg : matchingItinerary.legs) {
+                if (leg.transitLeg) {
+                    journeyState.originalDepartureTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                        leg.departureDelay,
+                        TimeUnit.SECONDS
+                    );
+                    break;
+                }
+            }
+            journeyState.originalArrivalTimeEpochMillis = matchingItinerary.endTime.getTime();
+            for (int i = matchingItinerary.legs.size() - 1; i >= 0; i--) {
+                Leg leg = matchingItinerary.legs.get(i);
+                if (leg.transitLeg) {
+                    journeyState.originalArrivalTimeEpochMillis -= TimeUnit.MILLISECONDS.convert(
+                        leg.arrivalDelay,
+                        TimeUnit.SECONDS
+                    );
+                    break;
+                }
+            }
+
             // save journey state with updated matching itinerary and target date
             journeyState.update(this);
         } else {
```

Comment:
I realize it's not quite in scope, but this else clause feels very naked.

Reply:
It does kinda come out of nowhere. Refactored in a8d90ab7a8ea11d1ed9c3c64e90e122e4a8a6d14